tests/test_cachito_util.py

```diff
@@ -0,0 +1,154 @@
+"""
+Copyright (c) 2019 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import absolute_import
+
+from atomic_reactor.cachito_util import (
+    CachitoAPI, CachitoAPIInvalidRequest, CachitpAPIRequestTimeout, CachitoAPIUnsuccessfulRequest)
+
+from requests.exceptions import HTTPError
+import pytest
+import responses
+import json
+import os.path
+
+
+CACHITO_URL = 'http://cachito.example.com'
+CACHITO_REQUEST_ID = 123
+CACHITO_REQUEST_REF = 'e1be527f39ec31323f0454f7d1422c6260b00580'
+CACHITO_REQUEST_REPO = 'https://github.com/release-engineering/retrodep.git'
+
+
+@responses.activate
+@pytest.mark.parametrize('additional_params', (
+    {},
+    {'flags': ['spam', 'bacon']},
+    {'pkg_managers': ['gomod']},
+    {'user': 'ham'},
+))
+def test_request_sources(additional_params):
+
+    def handle_request_sources(http_request):
+        body_json = json.loads(http_request.body)
+
+        assert body_json['repo'] == CACHITO_REQUEST_REPO
+        assert body_json['ref'] == CACHITO_REQUEST_REF
+        for key, value in additional_params.items():
+            assert body_json[key] == value
+
+        return (201, {}, json.dumps({'id': CACHITO_REQUEST_ID}))
+
+    responses.add_callback(
+        responses.POST,
+        '{}/api/v1/requests'.format(CACHITO_URL),
+        content_type='application/json',
+        callback=handle_request_sources)
+
+    api = CachitoAPI(CACHITO_URL)
+    response = api.request_sources(CACHITO_REQUEST_REPO, CACHITO_REQUEST_REF, **additional_params)
+    assert response['id'] == CACHITO_REQUEST_ID
+
+
+@responses.activate
+@pytest.mark.parametrize(('status_code', 'error', 'error_body'), (
+    (400, CachitoAPIInvalidRequest, json.dumps({'error': 'read the docs, please'})),
+    (500, HTTPError, 'Internal Server Error'),
+))
+def test_request_sources_error(status_code, error, error_body):
+    responses.add(
+        responses.POST,
+        '{}/api/v1/requests'.format(CACHITO_URL),
+        content_type='application/json',
+        body=error_body,
+        status=status_code,
+    )
+
+    with pytest.raises(error):
+        CachitoAPI(CACHITO_URL).request_sources(CACHITO_REQUEST_REPO, CACHITO_REQUEST_REF)
+
+
+@responses.activate
+@pytest.mark.parametrize('burst_params', (
+    {'burst_retry': 0.01, 'burst_length': 0.5, 'slow_retry': 0.2},
+    # Set the burst_retry to lower than burst_length to trigger the slow_retry :)
```

Comment:
Weird, coveralls is reporting that the slow retry is not being triggered here.

Reply:
That's because I swapped burst_retry with burst_length. We want the length to be shorter to trigger the slow_retry. This should be fixed now.